packages/client/lib/rpc/modules/engine.ts

```diff
@@ -173,6 +173,25 @@ const validHash = async (
   return bufferToHex(hash)
 }
 
+/**
+ *  Validate that the head block sent is a not a PoW block
+ */
+const validateTerminalBlock = async (block: Block, chain: Chain): Promise<boolean> => {
+  const ttd = chain.config.chainCommon.hardforkTD()
+  if (ttd === null) return false
+  const blockTd = await chain.getTd(block.hash(), block.header.number)
+  if (!block.header.parentHash.toString()) {
```

Comment:
awesome PR, nice work :) what's this if condition checking?

Reply:
It checks if the parent hash it's not `0x0..0` - This is not happening in mainnet but I thought that it's more for development and experimental purposes?

Reply:
hm that would be `block.header.parentHash.equals(Address.zero())`

maybe add a comment under the if statement that says `// This should only happen for development purposes, for when the genesis block has parentHash 0x000...000`

you could also probably also do `!block.isGenesis()`

Reply:
yes, for (dev) chains starting from a genesis block >=TTD, genesis block is the terminal block, the parentHash should be compared against zero hex, or as Ryan suggested !block.isGenesis